mlir/test/Quantum/ConversionTest.mlir

```diff
@@ -217,6 +217,59 @@ func.func @custom_gate(%q0 : !quantum.bit, %p : f64) -> (!quantum.bit, !quantum.
 
 // -----
 
+// CHECK-DAG: llvm.func @__catalyst__qis__RX(f64, !llvm.ptr, !llvm.ptr)
+// CHECK-DAG: llvm.func @__catalyst__qis__CRot(f64, f64, f64, !llvm.ptr, !llvm.ptr, !llvm.ptr)
+
+// CHECK-LABEL: @custom_gate_static_param
+func.func @custom_gate_static_param(%q0 : !quantum.bit) -> (!quantum.bit, !quantum.bit, !quantum.bit) {
+
+    // CHECK: [[one:%.+]] = llvm.mlir.constant(1 : i64) : i64
+    // CHECK: [[nullPtr:%.+]] = llvm.mlir.zero : !llvm.ptr
+    // CHECK: [[f:%.+]] = llvm.mlir.constant(2.000000e-01 : f64) : f64
+    // CHECK: llvm.call @__catalyst__qis__RX([[f]], %arg0, [[nullPtr]])
+    %q1 = quantum.custom "RX"( [2.000000e-01]) %q0 : !quantum.bit
+
+    // CHECK: [[one:%.+]] = llvm.mlir.constant(1 : i64) : i64
+    // CHECK: [[nullPtr:%.+]] = llvm.mlir.zero : !llvm.ptr
+    // CHECK: [[true:%.+]] = llvm.mlir.constant(true) : i1
+    // CHECK: [[zero:%.+]] = llvm.mlir.constant(0 : i64) : i64
+    // CHECK: [[struct:%.+]] = llvm.alloca [[one]] x !llvm.struct<(i1, i64, ptr, ptr)> : (i64) -> !llvm.ptr
+    // CHECK: [[bool_ptr:%.+]] = llvm.getelementptr inbounds [[struct]][0, 0] : (!llvm.ptr) -> !llvm.ptr, !llvm.struct<(i1, i64, ptr, ptr)>
+    // CHECK: [[int_ptr:%.+]] = llvm.getelementptr inbounds [[struct]][0, 1] : (!llvm.ptr) -> !llvm.ptr, !llvm.struct<(i1, i64, ptr, ptr)>
+    // CHECK: [[ptr1:%.+]] = llvm.getelementptr inbounds [[struct]][0, 2] : (!llvm.ptr) -> !llvm.ptr, !llvm.struct<(i1, i64, ptr, ptr)>
+    // CHECK: [[ptr2:%.+]] = llvm.getelementptr inbounds [[struct]][0, 3] : (!llvm.ptr) -> !llvm.ptr, !llvm.struct<(i1, i64, ptr, ptr)>
+    // CHECK: llvm.store [[true]], [[bool_ptr]] : i1, !llvm.ptr
+    // CHECK: llvm.store [[zero]], [[int_ptr]] : i64, !llvm.ptr
+    // CHECK: llvm.store [[nullPtr]], [[ptr1]] : !llvm.ptr, !llvm.ptr
+    // CHECK: llvm.store [[nullPtr]], [[ptr2]] : !llvm.ptr, !llvm.ptr
+    // CHECK: [[f2:%.+]] = llvm.mlir.constant(2.000000e-01 : f64) : f64
```

Comment:
Is this angle not negated by adjoint?

Reply:
I think this is not negated by adjoint because in `populateParameters` during conversion patterns there is no conditional on adjoint flag

Reply:
hmm, I am not too familiar with the effect of the adjoint but I believe the logic should be the same as before. If the angle should be negated and it is not, then it is an existing issue, because the same problem exists in the [dynamic test case](https://github.com/PennyLaneAI/catalyst/blob/f5f0aa4a65c5b0263f53a163e03b1f364c9a158c/mlir/test/Quantum/ConversionTest.mlir#L204).
I think we can deal with this problem in another PR, since this one is getting a bit big.

Reply:
I just realized the "adjoint negating the rotation angles" happens during canonicalization https://github.com/PennyLaneAI/catalyst/blob/main/mlir/lib/Quantum/IR/QuantumOps.cpp#L51 of the gate ops, which is probably before the merge rotation passes (and certainly before the quantum-dialect-to-llvm conversion, which is pretty much the last thing in the pipeline), so the adjoint attributes would never reach this stage.

Some confirmation would be nice!

Reply:
@paul0403, yes you are right. I confirmed that the negation happens at canonicalization stage right at the beginning.